app/scripts/modules/core/rollup.config.js

```diff
@@ -0,0 +1,42 @@
+import alias from '@rollup/plugin-alias';
+import fs from 'fs';
+import path from 'path';
+import copy from 'rollup-plugin-copy';
+
+import baseRollupConfig from '@spinnaker/scripts/config/rollup.config.base.module';
+import externalConfigurer from '@spinnaker/scripts/helpers/rollup-node-auto-external-configurer';
+
+const packageJSON = JSON.parse(fs.readFileSync('./package.json', 'utf8'));
+const externals = {
+  ...packageJSON.dependencies,
+  'root/version': true,
+};
+
+const fixTSPathRewrite = () => {
+  return {
+    name: 'fixTSPathRewrite',
+    writeBundle: () => {
+      const dts = path.join(__dirname, 'dist', 'index.d.ts');
+      const fixed = fs
+        .readFileSync(dts)
+        .toString()
+        .replace(/types="core\/types"/, 'types="./types"');
+      fs.writeFileSync(dts, fixed);
+    },
+  };
+};
+
+export default {
+  ...baseRollupConfig,
+  plugins: [
+    ...baseRollupConfig.plugins,
+    alias({
+      entries: [{ find: 'root', replacement: path.join(__dirname, '../../../../') }],
+    }),
+    copy({
+      targets: [{ src: 'src/types/**/*', dest: 'dist/types' }],
+    }),
```

Comment:
why does core need types but nothing else does?  For example titus depends on both the amazon and docker packages.

Reply:
I'm going to add some comments in this file.

Rollup uses the typescript plugin to automatically generate the type definitions of modules, however `core` contains some additional typescript declarations such as [`promise.d.ts`](https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/types/promise.d.ts) and [`svg.d.ts`](https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/types/svg.d.ts) that are needed when linking them to `deck-nflx`.